### PR TITLE
Add Velocity Template Language support

### PR DESCRIPTION
**Pull request by PieterO**
## Description

This should add support for [velocity](https://velocity.apache.org/) templating language.

> Velocity is a Java-based template engine. It permits anyone to use a simple yet powerful template language to reference objects defined in Java code.

## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3Avtl+NOT+nothack
  - [X] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [header.velocity](https://github.com/moleculer-java/moleculer-java-web/blob/a3ce980dac08f65e1ba41611f9e4eced55948772/src/test/java/www/header.velocity)
      - [index.vtl](https://codemirror.net/mode/velocity/index.html)
    - Sample license(s):
  
  - [x] I have included a syntax highlighting grammar: https://github-lightshow.herokuapp.com/
    - [Preview 1](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=cson&grammar_url=https%3A%2F%2Fgithub.com%2FAlhadis%2Flanguage-etc%2Fblob%2Fmaster%2Fgrammars%2Fneon.cson&grammar_text=&code_source=from-text&code_url=&code=name%3A+Homer++++%23+this+is+a+comment%0D%0A%0D%0Aaddress%3A%0D%0A%09street%3A+742+Evergreen+Terrace%0D%0A%09city%3A+Springfield%0D%0A%09country%3A+USA%0D%0A%0D%0Aphones%3A+%7B+home%3A+555-6528%2C+work%3A+555-7334+%7D%0D%0A%0D%0Achildren%3A%0D%0A%09-+Bart%0D%0A%09-+Lisa%0D%0A%09-+Maggie%0D%0A%0D%0Aentity%3A+Column%28type%3Dint%2C+nulls%3Dyes%29%0D%0A%0D%0Astring%3A+%27quotes+string%27%0D%0A%0D%0Amulti+line+string%3A+%27%27%27%0D%0A%09one+line%0D%0A%09second+line%0D%0A%09third+line%0D%0A%27%27%27%0D%0A%0D%0Aspecial+types%3A%0D%0A%09escape+sequences%3A+%22%5Cu2026+%5Ct+%5Cn+%5Cr+%5Cf+%5Cb+%5Cx65%22%0D%0A%09bool%3A+%5Btrue%2C+false%5D%0D%0A%09bool+alternative%3A+%5Byes%2C+no%5D%0D%0A%09exponent+number%3A+%2B0123.45e6%0D%0A%09hex+number%3A+0xAB01%0D%0A%09octal+number%3A+0o666%0D%0A%09binary+number%3A+0b11100111%0D%0A%09date%3A+2020-02-02%0D%0A%09date+time%3A+2020-02-02+12%3A34%3A56%0D%0A)
    - [Preview 2](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=cson&grammar_url=https%3A%2F%2Fgithub.com%2FAlhadis%2Flanguage-etc%2Fblob%2Fmaster%2Fgrammars%2Fneon.cson&grammar_text=&code_source=from-text&code_url=&code=name%3A+Homer++++%23+this+is+a+comment%0D%0A%0D%0Aaddress%3A%0D%0A%09street%3A+742+Evergreen+Terrace%0D%0A%09city%3A+Springfield%0D%0A%09country%3A+USA%0D%0A%0D%0Aphones%3A+%7B+home%3A+555-6528%2C+work%3A+555-7334+%7D%0D%0A%0D%0Achildren%3A%0D%0A%09-+Bart%0D%0A%09-+Lisa%0D%0A%09-+Maggie%0D%0A%0D%0Aentity%3A+Column%28type%3Dint%2C+nulls%3Dyes%29%0D%0A%0D%0Astring%3A+%27quotes+string%27%0D%0A%0D%0Amulti+line+string%3A+%27%27%27%0D%0A%09one+line%0D%0A%09second+line%0D%0A%09third+line%0D%0A%27%27%27%0D%0A%0D%0Aspecial+types%3A%0D%0A%09escape+sequences%3A+%22%5Cu2026+%5Ct+%5Cn+%5Cr+%5Cf+%5Cb+%5Cx65%22%0D%0A%09bool%3A+%5Btrue%2C+false%5D%0D%0A%09bool+alternative%3A+%5Byes%2C+no%5D%0D%0A%09exponent+number%3A+%2B0123.45e6%0D%0A%09hex+number%3A+0xAB01%0D%0A%09octal+number%3A+0o666%0D%0A%09binary+number%3A+0b11100111%0D%0A%09date%3A+2020-02-02%0D%0A%09date+time%3A+2020-02-02+12%3A34%3A56%0D%0A)

  - [ ] ~~I have updated the heuristics to distinguish my language from others using the same extension.~~
     - I don't think there are any extension collisions so probably alright.

  
  - [ ] I have obtained agreement from the wider language community on this color change.
    - I tried to make it the same colour as the logo (`#37379b`, `#2a497e` or `#093a6b`)  so settled for `#507cff` instead . Can ask around to get it right.

### CODE REVIEW

**Comment by smola** [comment]
Bump.

**Comment by Nixinova** [comment]
Is this ready to undraft?

**Comment by PieterO** [comment]
> Is this ready to undraft?

Only took me a couple of centuries :laughing:, I think this should cover it but it was a while so I'll recheck the template/actions will need rerunning.

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -6786,6 +6786,20 @@ Valve Data Format:
...
   language_id: 544060961
+Velocity Template Language:
+  type: markup
+  color: "#507cff"
+  aliases:
+  - vtl
+  - velocity
+  extensions:
+  - ".vtl"
+  - ".velocity"
```
Comment (lildude) [inline lib/linguist/languages.yml:6797]: Please remove this extension. I've not included any samples for it and it's nowhere near popular enough yet.

File: samples/Velocity Template Language/hello-world.vtl [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    @author John Doe
+    @version 1
+*#
+
+<html>
+  <body>
+  #set( $foo = "Hello" )
+   $foo World! ## This would print, you guessed it, Hello World!
+  </body>
+</html>
```
Comment (lildude) [inline samples/Velocity Template Language/hello-world.vtl:12]: Please remove/replace this sample. We explicitly ask for real-world samples as "hello worlds" are never really not generally illustrative of a language

**Review by lildude — CHANGES_REQUESTED** [review]

**Review by lildude — APPROVED** [review]

File: lib/linguist/languages.yml (hunk 1 above)
Comment (Zerocoolooog) [reply lib/linguist/languages.yml]: > Please remove this extension. You've not included any samples for it and it's nowhere near popular enough yet.

098f77c7c9158095b402916a6cdb1ee3307dd2ef

**Pull request merged**